src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -319,6 +322,57 @@ CreateReferenceTableColocationId()
 
 
 /*
+ * DeleteAllReferenceTablePlacementsFromNode function iterates over list of reference
+ * tables and deletes all reference table placements from pg_dist_shard_placement table
+ * for given worker node. It also modifies replication factor of the colocation group of
+ * reference tables so that replication factor will be equal to number of workers.
+ */
+void
+DeleteAllReferenceTablePlacementsFromNode(char *workerName, uint32 workerPort)
+{
+	List *referenceTableList = ReferenceTableOidList();
+	ListCell *referenceTableCell = NULL;
+
+	Oid firstReferenceTableId = InvalidOid;
+	uint32 referenceTableColocationId = INVALID_COLOCATION_ID;
+
+	List *workerNodeList = NIL;
+	int workerCount = 0;
+
+	/* if there is no reference table, we do not need to do anything */
+	if (list_length(referenceTableList) == 0)
+	{
+		return;
+	}
+
+	referenceTableList = SortList(referenceTableList, CompareOids);
+	foreach(referenceTableCell, referenceTableList)
+	{
+		Oid referenceTableId = lfirst_oid(referenceTableCell);
+
+		List *shardIntervalList = LoadShardIntervalList(referenceTableId);
+		ShardInterval *shardInterval = (ShardInterval *) linitial(shardIntervalList);
+		uint64 shardId = shardInterval->shardId;
+
+		LockShardDistributionMetadata(shardId, ExclusiveLock);
+
+		DeleteShardPlacementRow(shardId, workerName, workerPort);
+	}
+
+	/*
+	 * After deleting reference tables placements, we will update replication factor
+	 * column for colocation group of reference tables so that replication factor will
+	 * be equal to worker count again.
+	 */
+	workerNodeList = WorkerNodeList();
```

Comment:
Note that this line relies on `DeleteNodeRow()` being called before calling this function. So, if you move `DeleteAllReferenceTablePlacementsFromNode ()` you need to make sure that we update colocation information with the correct worker count.